engine/api/workflow/dao_workflow_integration.go

```diff
@@ -0,0 +1,66 @@
+package workflow
+
+import (
+	"context"
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/engine/api/integration"
+	"github.com/ovh/cds/sdk"
+)
+
+// AddWorkflowIntegration link a project integration on a workflow
+func AddWorkflowIntegration(db gorp.SqlExecutor, integ *sdk.WorkflowProjectIntegration) error {
+	dbInteg := dbWorkflowProjectIntegration(*integ)
+	if err := gorpmapping.Insert(db, &dbInteg); err != nil {
+		return err
+	}
+	integ.ID = dbInteg.ID
+	return nil
+
+}
+
+// LoadWorkflowIntegrationsByWorkflowID load workflow integrations by workflowid
+func LoadWorkflowIntegrationsByWorkflowID(db gorp.SqlExecutor, id int64) ([]sdk.WorkflowProjectIntegration, error) {
+	query := gorpmapping.NewQuery(`
+		SELECT workflow_project_integration.*
+		FROM workflow_project_integration
+		WHERE workflow_project_integration.workflow_id = $1
+	`).Args(id)
+	return loadAll(db, query)
+}
+
+// RemoveIntegrationFromWorkflow remove a project integration on a workflow
+func RemoveIntegrationFromWorkflow(db gorp.SqlExecutor, workflowID int64, projectIntegrationID int64) error {
+	query := "DELETE FROM workflow_project_integration WHERE workflow_id = $1 AND project_integration_id = $2"
+	if _, err := db.Exec(query, workflowID, projectIntegrationID); err != nil {
+		return sdk.WithStack(err)
+	}
+	return nil
+}
+
+// DeleteIntegrationsFromWorkflow remove a project integration on a workflow
+func DeleteIntegrationsFromWorkflow(db gorp.SqlExecutor, workflowID int64) error {
+	query := "DELETE FROM workflow_project_integration WHERE workflow_id = $1"
+	if _, err := db.Exec(query, workflowID); err != nil {
```

Comment:
is it possible to use gorpmapping.Delete ?

Reply:
no because I'm currently deleting all integrations linked to the workflow